lib/spack/spack/detection/common.py

```diff
@@ -168,7 +168,8 @@ def executable_prefix(executable_dir: str) -> str:
     if "bin" not in lowered_components:
         return executable_dir
     idx = lowered_components.index("bin")
-    return os.sep.join(components[:idx])
+    # Do not return an empty string if "bin" is found in the root directory
+    return os.sep.join(components[:idx]) if idx > 1 else os.path.abspath(os.sep)
```

Comment:
This conditional is repeated 4 times, and is more complex than the previous computation. Can we extract it is a function of its own?

Reply:
I gave it a try. Let me know if that test is sufficient.